File: Client/Desktop/src/main/java/com/example/tin/security/Encryption.java
package com.example.tin.security;

import java.util.Arrays;
import java.util.Base64;

public class Encryption {
    String key = "ABC123EF";
    Aes256 aes = new Aes256();

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String encrypt(String message)
    {
        try {
            System.out.println("Szyfruję wiadomosc: ");
            System.out.println(message);
            String encrypted = aes.encrypt(message, key);
            System.out.println("Szyfrogram: ");
            System.out.println(encrypted);
            return encrypted;
        } catch( Exception e) {
            e.printStackTrace();
            return "";
        }
    }
/*
    public String decrypt(String message)
    {
        try {
            String decrypted =  aes.decrypt(message, key);
            System.out.println("Odszyfrowana wiadomosc: " + decrypted);
            return decrypted;


        } catch (Exception e) {
            e.printStackTrace();
            return "";
        }
    }
*/

    public String decrypt(String message)
    {
        final byte[] inBytes = Base64.getDecoder().decode(message);
        message = inBytes.toString();
        System.out.println("Po basie: " + Arrays.toString(inBytes));
        StringBuilder sb = new StringBuilder();
        for(int i = 0; i < inBytes.length; i++)
            sb.append((char)(inBytes[i] ^ key.charAt(i % (key.length()))));
        String res = sb.toString();
        System.out.println("Wiadomość odszyfrowana: " + res);
        return(res);
    }


}

File: Server/threads/Encryption.cpp
//
// Created by root on 6/2/19.
//

#include "Encryption.h"
#include "security/base64.h"
#include <iostream>

/*
const string &Encryption::getKey() const {
    return key;
}

void Encryption::setKey(const string &key) {
    Encryption::key = key;
} */

string Encryption::encrypt(string toEncrypt, string key) {

    string output = toEncrypt;
    for (int i = 0; i < toEncrypt.length(); i++)
        output[i] = toEncrypt[i] ^ key[i % (sizeof(key) / sizeof(char))];
    return base64_encode((const unsigned char*)output.c_str(), output.length());
}

string Encryption::decrypt(string message, string key) {
    return aes.decrypt(message, key);
}


